jskit/encoders/bi.py

```diff
@@ -101,16 +105,40 @@ def cosine_sim(vec_a: List[float], vec_b: List[float], meta):
 
 
 @jaseci_action(act_group=['bi_enc'], allow_remote=True)
-def infer(contexts: List, candidates: List):
+def infer(contexts: Union[List[str], List[List[float]]], candidates: Union[List[str], List[List[float]]]):
     """
     Take list of context, candidate and return nearest candidate to the context
     """
     global model
     model.eval()
-    predicted_candidate = get_inference(model, tokenizer,
-                                        contexts=contexts,
-                                        candidates=candidates)
-    return predicted_candidate
+    predicted_candidates = []
+    if is_list_of(data=contexts, dtype=str) and is_list_of(data=candidates, dtype=str):
+        predicted_candidates = get_inference(model, tokenizer,
+                                             contexts=contexts,
+                                             candidates=candidates)
+    elif is_list_of(data=contexts, dtype=str) and is_list_of(data=candidates, dtype=list):
+        con_embed = []
+        con_embed = get_candidate_emb(contexts)
+        for data in con_embed:
+            score_dat = []
+            for lbl in candidates:
+                score_dat.append(cosine_sim(
+                    vec_a=data, vec_b=lbl, meta="string"))
+            predicted_candidates.append(candidates[np.argmax(score_dat)])
+    elif is_list_of(data=contexts, dtype=list) and is_list_of(data=candidates, dtype=str):
+        cand_embed = []
+        predicted_candidates = []
+        cand_embed = get_candidate_emb(candidates)
+        for data in contexts:
+            score_dat = []
+            for lbl in cand_embed:
+                score_dat.append(cosine_sim(
+                    vec_a=data, vec_b=lbl, meta="string"))
+            predicted_candidates.append(candidates[np.argmax(score_dat)])
+    else:
+        raise HTTPException(status_code=404, detail=str(
+            "Invalid argument"))
```

Comment:
With the explicit function parameter, you can also make this error message more detailed.